src/models/tournament/game.controller.ts

```diff
@@ -0,0 +1,74 @@
+import { BadRequestException, Body, Controller, Get, Logger, NotImplementedException, Post, Put, ValidationPipe } from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { DefaultExceptionDto } from './dto/error.dto';
+import { GameUpdateDto } from './dto/gameUpdate.dto';
+import { CreateGameDto, GameDto, GameQuery } from './dto/game.dto';
+import { GameService } from './services/game.service';
+
+@ApiTags('Tournaments', 'TournamentGame')
+@Controller({
+	path: 'tournaments/game',
+	version: ['1'],
+})
+export class GameController {
+	private readonly logger = new Logger(GameController.name);
+
+	constructor(private readonly gameService: GameService) {}
+
+	@ApiOperation({ operationId: 'game_update', summary: 'Receives updates (e.g. from playtak-server) about games played' })
+	@ApiResponse({
+		status: 200,
+		type: null,
+		description: 'Confirmation that the update has been received',
+	})
+	@ApiResponse({
+		status: 500,
+		type: DefaultExceptionDto,
+		description: 'Returns 500 server error',
+	})
+	@ApiResponse({
+		status: 400,
+		type: BadRequestException,
+	})
+	@ApiResponse({
+		status: 501,
+		type: NotImplementedException,
+		description: 'When called with arguments not yet handled',
+	})
+	@Post("update")
+	gameUpdate(@Body() gameUpdate: GameUpdateDto): Promise<void|NotImplementedException|TypeError> {
+		// TODO: This should only allow requests from playtak - is it enough to check for a localhost IP?
+		//       or do we need a secret token?
+
+		if (!gameUpdate?.game) {
```

Comment:
is there a reason this check is in the controller vs the service handleGameUpdate method?

Reply:
I felt like the controller should be the one throwing HttpExceptions and checking the input data and the service could expect validated data, but currently the service is doing both in most cases anyways so I might as well move this check over to the controller.

Reply:
done